### PR TITLE
Fix Broken Index Shard Snapshot File Preventing Snapshot Creation

### PR DESCRIPTION
**Pull request by mkleen**
Backport of https://github.com/elastic/elasticsearch/pull/41310

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by mkleen** [comment]
We removed `SharedClusterSnapshotRestoreIT` from our tests, and not 100 % if i can port the related test over here.

**Comment by mfussenegger** [comment]
> We removed `SharedClusterSnapshotRestoreIT` from our tests, and not 100 % if i can port the related test over here.

I think using SQL to create the snapshot + queries on sys tables  it should be possible to create a test case

**Comment by mkleen** [comment]
> > We removed `SharedClusterSnapshotRestoreIT` from our tests, and not 100 % if i can port the related test over here.
> 
> I think using SQL to create the snapshot + queries on sys tables it should be possible to create a test case

Done.

File: server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java [hunk 1]
```diff
@@ -485,4 +493,86 @@ public void test_cannot_create_snapshot_in_read_only_repo() {
...
+        var numberOfDocs = randomLongBetween(0, 10);
+        for (int i = 0; i < numberOfDocs; i++) {
+            execute("INSERT INTO t1 (x) VALUES (?)", new Object[]{randomInt()});
+        }
+        execute("REFRESH TABLE t1");
+
+        var snapShotName1 = "s1";
+        var fullSnapShotName1 =  REPOSITORY_NAME + "." + snapShotName1;
+        execute("CREATE SNAPSHOT " + fullSnapShotName1 + " ALL WITH (wait_for_completion=true)");
+        ensureYellow();
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java:508]:
```suggestion
```

Shouldn't be required here.

File: server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java [hunk 2]
```diff
@@ -485,4 +493,86 @@ public void test_cannot_create_snapshot_in_read_only_repo() {
...
+        for (int i = 0; i < numberOfAdditionalDocs; i++) {
+            execute("INSERT INTO t1 (x) VALUES (?)", new Object[]{randomInt()});
+        }
+        execute("REFRESH TABLE t1");
+
+        var snapShotName2 = "s2";
+        var fullSnapShotName2 = REPOSITORY_NAME + ".s2";
+
+        execute("CREATE SNAPSHOT " + fullSnapShotName2 + " ALL WITH (wait_for_completion=true)");
+        ensureYellow();
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/SnapshotRestoreIntegrationTest.java:543]:
```suggestion
```

**Pull request merged**